enhancements/windows-containers/csi-proxy.md

```diff
@@ -132,34 +132,111 @@ No action required.
-of volumes when upgrading from OCP 4.12 and 4.13. This specifically effects Azure, which loses in-tree storage
-functionality in 4.13, and vSphere which loses it in 4.14. New clusters on other platforms should be created with a
-minumum version of OCP 4.12, so they can immediately use CSI driver storage, as there will not be a way to migrate
-in-tree volumes for those platforms.
+##### Supported upgrades
+WMCO will only support upgrading in-tree storage workloads to CSI driver storage workloads for Azure and vSphere.
+New cluster installs on other platforms should be created with a minumum version of OCP 4.12, so they can immediately
+use CSI driver storage, as there will not be a way to safely migrate in-tree volumes for those platforms. This is due
+to a lack of Windows in-tree storage support leading up to 4.12 for those platforms.
+
+##### Node upgrades
+Cluster upgrades will have an impact on this feature when upgrading from a cluster version which uses in-tree
+storage, to one which uses CSI drivers. [When upgrading cluster versions, in-tree driver Volumes will automatically
+be updated to use the appropriate CSI node drivers by the cluster](https://docs.openshift.com/container-platform/4.12/storage/container_storage_interface/persistent-storage-csi-migration.html).
+Volumes attached to Windows Nodes, will remain usable until the Node's kubelet is upgraded to the version matching the
+new cluster version. This overlap period will allow for a smooth transition for each Node.
+
+In order to properly make use of this, and ensure a clean transition with minimal workload interruption, both csi-proxy
+and the CSI node drivers must be running on the Windows node before the upgrade commences. It will be up to the cluster
+administrator to install the CSI drivers on the cluster. They should check that all deployed pods go into running, and
+a CSINode is created for each Windows node. The CSINode should list the CSI driver as enabled.
+
+The requirement of csi-proxy is one that must be fufilled by WMCO. As this feature is targeted for WMCO 9.0.0,
+csi-proxy must be backported to WMCO v7 and v8, to enable a smooth transition of volumes when upgrading from OCP
+4.12 and 4.13. This specifically effects Azure, which loses in-tree storage functionality in 4.13, and vSphere which
+loses it in 4.14.
+
+What this means for the user, is that when they are upgrading a cluster from a version that uses in-tree to one that
+uses CSI, they must first upgrade WMCO to a version which runs csi-proxy on Windows Nodes. Only then should they
+upgrade the cluster.
+
+In order to help ensure a smooth transition WMCO v8 and v9 will block each individual Node upgrade if:
+1) A Node is being upgraded from in-tree storage to CSI driver storage
+2) A volume is attached to the Node
+
+The cluster administrator will need to apply the label `windowsmachineconfig.openshift.io/allow-upgrade=true` to each
```

Comment:
it may be late for this, but what about `windowsmachineconfig.openshift.io/upgradeable=true` for the label?

Reply:
I'd prefer to keep it as `allow-upgrade` as it emphasizes more that the user is the one allowing the upgrade to occur, instead of the Node meeting all possible conditions to be upgraded.